taichi/math/arithmetic.h

```diff
@@ -14,4 +14,19 @@ T iroundup(T a, T b) {
   }
 }
 
+template <typename T>
+uint32_t log2int(T value) {
+  if constexpr (!std::is_integral_v<T>) {
+    static_assert(always_false_v<T>, "Must be integral type");
+  }
```

Comment:
Why not just
```suggestion
  static_assert(std::is_integral_v<T>, "Must be integral type");
```
? Sorry, I'm not a C++ expert..